crates/red_knot_python_semantic/src/types.rs

```diff
@@ -355,6 +355,13 @@ impl<'db> Type<'db> {
     pub const fn into_int_literal_type(self) -> Option<i64> {
         match self {
             Type::IntLiteral(value) => Some(value),
+            Type::BooleanLiteral(b) => {
+                if b {
+                    Some(1)
+                } else {
+                    Some(0)
+                }
+            }
```

Comment:
We should not add this; boolean literals are not equivalent to int literals, the conversion should not be implicit. See below comment for how we can handle the conversion explicitly without doing this.